blockchain/blockchain_test.go

```diff
@@ -1644,3 +1659,400 @@ func TestBlockchain_WriteFullBlock(t *testing.T) {
+	require.NoError(t, err)
+
+	receiptsByteVaulue, _ := io.ReadAll(receiptsJSONFile)
+
+	blockWriter(t, checkInterval, blockTime, numberOfBlocks, byteValue, receiptsByteVaulue)
+}
+
+func TestDiskUsageWriteBatchAndUpdateNoTimeout(t *testing.T) {
+	const (
+		checkInterval  = 15 // milliseconds
+		numberOfBlocks = 100
+		blockTime      = 0 //milliseconds
+	)
+
+	jsonFile, err := os.Open("testblock.json")
+	require.NoError(t, err)
+
+	byteValue, _ := io.ReadAll(jsonFile)
+
+	receiptsJSONFile, err := os.Open("receipts.json")
+	require.NoError(t, err)
+
+	receiptsByteVaulue, _ := io.ReadAll(receiptsJSONFile)
+
+	blockWriter(t, checkInterval, blockTime, numberOfBlocks, byteValue, receiptsByteVaulue)
+}
+
+func blockWriter(t *testing.T, checkInterval, blockTime, numberOfBlocks uint64, byteToRead []byte, receiptsBytesToRead []byte) {
+	t.Helper()
+
+	type blockCounter struct {
+		mu sync.RWMutex
+		x  uint64
+	}
+
+	counter := &blockCounter{x: 0}
+
+	quitChan := make(chan bool)
+
+	p, err := os.MkdirTemp("", "DiskUsageTest")
+	require.NoError(t, err)
+
+	db, err := leveldb.NewLevelDBStorage(
+		filepath.Join(p),
+		hclog.NewNullLogger(),
+	)
+	require.NoError(t, err)
+
+	block, err := CustomJSONBlockUnmarshall(byteToRead)
+	require.NoError(t, err)
+
+	receipts, err := CustomReceiptsUnmarshall(receiptsBytesToRead)
+	require.NoError(t, err)
+
+	dirSizeCheck := func() {
+		for {
+			select {
+			case <-quitChan:
+				return
+			default:
+				dirSizeValue, err := DirSize(p)
+
+				if err != nil {
+					t.Log(err)
+				}
+
+				counter.mu.RLock()
+
+				t.Logf("BLOCK: %d DIRSIZE IS: %d and average is:%.2f", counter.x, dirSizeValue, float64(dirSizeValue)/float64(counter.x))
+
+				counter.mu.RUnlock()
+
+				time.Sleep(time.Millisecond * time.Duration(checkInterval))
+			}
+		}
+	}
+
+	blockchain := &Blockchain{db: db}
+
+	require.NoError(t, os.MkdirAll(p, 0755))
+
+	require.NoError(t, err)
+
+	dirSizeBeforeBlocks, err := DirSize(p)
+	require.NoError(t, err)
+	t.Logf("DIRSIZE IS: %d", dirSizeBeforeBlocks)
+
+	go dirSizeCheck()
+
+	for i := 0; i < int(numberOfBlocks); i++ {
+		batchWriter := storage.NewBatchWriter(db)
+		block.Block.Header.Number = uint64(i)
+
+		batchWriter.PutHeader(block.Block.Header)
+		batchWriter.PutBody(block.Block.Hash(), block.Block.Body())
+
+		batchWriter.PutReceipts(block.Block.Hash(), receipts)
+
+		require.NoError(t, blockchain.writeBatchAndUpdate(batchWriter, block.Block.Header, big.NewInt(0), false))
+
+		counter.mu.Lock()
+
+		counter.x++
+
+		counter.mu.Unlock()
+
+		time.Sleep(time.Millisecond * time.Duration(blockTime))
+	}
+	quitChan <- true
+
+	dirSizeAfterBlocks, err := DirSize(p)
+	require.NoError(t, err)
+	t.Logf("DIRSIZE After All blocks: %d, average blocks size: %.2f", dirSizeAfterBlocks, float64(dirSizeAfterBlocks)/float64(numberOfBlocks))
+
+	db.Close()
+
+	assert.NotEqual(t, dirSizeBeforeBlocks, dirSizeAfterBlocks)
+}
+
+func CustomJSONBlockUnmarshall(jsonData []byte) (*types.FullBlock, error) {
```

Comment:
Don't expose the function and pass `testing.T` to it, and add `t.Helper()` at the beginning of the function.